pyrsa/rdm/calc.py

```diff
@@ -130,24 +141,24 @@ def calc_rdm_mahalanobis(dataset, descriptor=None, noise=None):
         pyrsa.rdm.rdms.RDMs: RDMs object with the one RDM
 
     """
-    measurements, desc, descriptor = _parse_input(dataset, descriptor)
-    noise = _check_noise(noise, dataset.n_channel)
-    diff = _calc_pairwise_differences(measurements)
-    diff2 = (noise @ diff.T).T
-    rdm = np.einsum('ij,ij->i', diff, diff2) / measurements.shape[1]
-    rdm = RDMs(dissimilarities=np.array([rdm]),
-               dissimilarity_measure='Mahalanobis',
-               descriptors=dataset.descriptors)
-    rdm.pattern_descriptors[descriptor] = desc
-    rdm.descriptors['noise'] = noise
+    if noise is None:
+        rdm = calc_rdm_euclid(dataset, descriptor)
+    else:
+        measurements, desc, descriptor = _parse_input(dataset, descriptor)
+        noise = _check_noise(noise, dataset.n_channel)
+        diff = _calc_pairwise_differences(measurements)
+        diff2 = (noise @ diff.T).T
+        rdm = np.einsum('ij,ij->i', diff, diff2) / measurements.shape[1]
+        rdm = RDMs(dissimilarities=np.array([rdm]),
+                   dissimilarity_measure='Mahalanobis',
+                   descriptors=dataset.descriptors)
+        rdm.pattern_descriptors[descriptor] = desc
+        rdm.descriptors['noise'] = noise
     return rdm
 
 
-def calc_rdm_crossnobis(dataset,
-                        descriptor,
-                        noise=None,
-                        cv_descriptor=None
-                        ):
+def calc_rdm_crossnobis(dataset, descriptor, noise=None,
+                        cv_descriptor=None, subversion='whiten'):
     """
```

Comment:
I think it rightly warns here about the CC of 34 I think. It looks like this is currently tested with this:

```
    def test_calc_crossnobis(self):
        rdm = rsr.calc_rdm_crossnobis(self.test_data, descriptor = 'conds', cv_descriptor = 'fold')
        assert rdm.n_cond == 6
```

So this probably only tests one path through the function. Just think of all the emails you're gonna get "I did this and then I got this error, what's wrong?"

Reply:
That is a good point. I will write a few more tests now to get at least a bit more covered

Reply:
now I added some more tests. I will now click this as resolved, but if you have ideas how to make the crossnobis easier I think this would be interesting still